src/nodes.coffee

```diff
@@ -2140,6 +2169,9 @@ exports.Code = class Code extends Base
     wasEmpty = @body.isEmpty()
     @body.expressions.unshift thisAssignments... unless @expandCtorSuper thisAssignments
     @body.expressions.unshift exprs...
+    if @isMethod and @bound and not @isStatic and @parentClass
```

Comment:
the runtime check, basically `_boundMethodCheck(this, parentClassRef)`. Will only be added for child class bound methods